pkg/manager/manager.go

```diff
@@ -52,9 +54,14 @@ func (m *Manager) ID() ID {
 	return m.id
 }
 
-// Config returns the configuration of the manager.
-func (m *Manager) Config() managercfg.Config {
-	return m.config
+// GetKubeconfig returns the Kubernetes REST config object associated with the instance.
+func (m *Manager) GetKubeconfig() *rest.Config {
+	return m.manager.GetKubeconfig()
+}
+
+// GetClientsManager returns the clients manager associated with the instance.
+func (m *Manager) GetClientsManager() *clients.AdminAPIClientsManager {
```

Comment:
We shouldn't return an internal struct here. As I can see `SetupAnonymousReports` needs this to calculate kongVersion, DB mode and router flavor. How about leaving a TODO here to change this getter to return an interface allowing fetching only this data? Or just have a getter on `Manager` that will return a struct with this data computed, WDYT?

Reply:
Very good point. I've added TODO. I want to rework `SetupAnonymousReports` to make it easier to use and the aforementioned is the thing that should be improved too. thx